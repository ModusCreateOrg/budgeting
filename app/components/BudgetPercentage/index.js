// @flow
import * as React from 'react';
import formatPercentage from 'utils/formatPercentage';
import type { Transaction } from 'modules/transactions';
import type { Categories } from 'modules/categories';
import styles from './style.scss';
import DonutChart from 'components/DonutChart';

type BudgetPercentageProps = {
  transaction: Transaction,
  inflows: number,
};
function createData(transaction,percent){
    var data = [];
    var item1 = {
        name : transaction.description,
        value : Math.abs(percent),
        id : 1,
        type: "percentage"
    };
    data.push(item1);
    var item2 = {
        name : "Remaining",
        value : 100-Math.abs(percent),
        id : 2,
        type: "percentage"
    };
    data.push(item2);
    return data;
}
const BudgetPercentage = ({ transaction, inflows }:BudgetPercentageProps) => {
    
  const { id, categoryId, description } = transaction;
  const percent = formatPercentage(transaction.value,inflows,categoryId);
  const percentCls = percent.isNegative ? styles.neg : styles.pos;
  var data = createData(transaction,percent.value);
  return (
      <div>
        <h3 className={percentCls}>{percent.text}</h3>
        <DonutChart data={data} dataLabel="name" dataKey="id"/>
    </div>
  );
};

export default BudgetPercentage;
