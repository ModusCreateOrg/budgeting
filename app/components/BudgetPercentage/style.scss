@import 'theme/variables';

.neg {
    color: $red;
  }
  
  .pos {
    color: $green;
  }