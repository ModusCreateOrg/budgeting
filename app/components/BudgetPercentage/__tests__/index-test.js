import React from 'react';
import renderer from 'react-test-renderer';
import BudgetPercentage from '../';

// mock nested components
jest.mock('../../DonutChart', () => 'div');

it('renders correctly', () => {
  const mockData = {
       description : "one",
    value : 40,
    id : 1
 };

  const tree = renderer
    .create(<BudgetPercentage transaction={mockData} inflows={100} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
