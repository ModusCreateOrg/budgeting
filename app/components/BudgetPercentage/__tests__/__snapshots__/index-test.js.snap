// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h3
    className="neg"
  >
    -40%
  </h3>
  <div
    data={
      Array [
        Object {
          "id": 1,
          "name": "one",
          "type": "percentage",
          "value": 40,
        },
        Object {
          "id": 2,
          "name": "Remaining",
          "type": "percentage",
          "value": 60,
        },
      ]
    }
    dataKey="id"
    dataLabel="name"
  />
</div>
`;
